torch_geometric/nn/aggr/basic.py

```diff
@@ -86,7 +113,11 @@ def forward(self, x: Tensor, index: Optional[Tensor] = None,
         alpha = x
         if not isinstance(self.t, (int, float)) or self.t != 1:
             alpha = x * self.t
-        alpha = softmax(alpha, index, ptr, dim_size, dim)
+        if not self.learn and self.semi_grad:
```

Comment:
```suggestion
        if not self.learn:
```
Related to above comment - could we just do this?

Reply:
I think this is correct. Either in case `t` is not learnable, we still may want to backprop through it.

Reply:
Right makes sense